src/Microsoft.NetCore.Analyzers/Core/Security/SetViewStateUserKey.cs

```diff
@@ -0,0 +1,101 @@
+            nameof(SystemSecurityCryptographyResources.SetViewStateUserKey),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Message = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.SetViewStateUserKeyMessage),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.SetViewStateUserKeyDescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                var compilation = compilationStartAnalysisContext.Compilation;
+                var pageTypeSymbol = compilation.GetTypeByMetadataName(WellKnownTypeNames.SystemWebUIPage);
+
+                if (pageTypeSymbol == null)
+                {
+                    return;
+                }
+
+                var eventArgsTypeSymbol = compilation.GetTypeByMetadataName(WellKnownTypeNames.SystemEventArgs);
+
+                if (eventArgsTypeSymbol == null)
+                {
+                    return;
+                }
+
+                compilationStartAnalysisContext.RegisterSymbolAction(symbolAnalysisContext =>
+                {
+                    var classSymbol = (INamedTypeSymbol)symbolAnalysisContext.Symbol;
+                    var baseClassSymbol = classSymbol.BaseType;
+
+                    if (baseClassSymbol.Equals(pageTypeSymbol))
+                    {
+                        var onInitMethodSymbol = classSymbol.GetMembers().OfType<IMethodSymbol>().FirstOrDefault(s => s.Name == "OnInit" &&
```

Comment:
What about the `Page_Init` method?

Reply:
It seems there's no method named Page_Init but OnInit.

Reply:
Oh, I guess `Page_Init` is usually the method that's the event handler for the `Control.Init` (https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.init) event.  It's not a virtual or abstract method.

I think we should look at it here too.